### PR TITLE
fix(combo): findMatch to return false when no item

### LINKED ISSUE #10726: Combo throws error if value is null or undefined
## Description 
Combo throws error when  value is null or undefined

Describe the issue.
In combo.common.ts
`protected findMatch = (element: any): boolean => {
        const value = this.displayKey ? element[this.displayKey] : element;
        return value.toString().toLowerCase() === this.searchValue.trim().toLowerCase();
    };`

value might be null or undefined and will throw an error at `value.toString()`
 * igniteui-angular version: 13.0.x

## Result  
image

## Attachments  
Attach a sample if available, and screenshots, if applicable.

### PR DESCRIPTION
**Pull request by Volen99**
Closes #10726

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by kdinev** [comment]
@Volen99 Please make your PR title according to the commit message convention, because otherwise it will go out in the changelog just as you have it right now.

**Comment by Volen99** [comment]
> @Volen99 Please make your PR title according to the commit message convention, because otherwise it will go out in the changelog just as you have it right now.

I am writing a test right now and with its commit will fix the commit message.

**Comment by kdinev** [comment]
> > @Volen99 Please make your PR title according to the commit message convention, because otherwise it will go out in the changelog just as you have it right now.
> 
> I am writing a test right now and with its commit will fix the commit message.

That is fine, however, I'm talking about the PR title, not the actual commit message.

**Review by jackofdiamond5 — DISMISSED** [review]

**Review by kdinev — DISMISSED** [review]

File: projects/igniteui-angular/src/lib/combo/combo.common.ts [hunk 1]
```diff
@@ -1183,6 +1183,9 @@ export abstract class IgxComboBaseDirective extends DisplayDensityBase implement
 
     protected findMatch = (element: any): boolean => {
         const value = this.displayKey ? element[this.displayKey] : element;
+        if (value === undefined || value === null) {
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/combo/combo.common.ts:1186]: why not use not null assertion operator like this:
```
return value?.toString().toLowerCase() === this.searchValue.trim().toLowerCase();
```

**Review by hanastasov — APPROVED** [review]

**Review by wnvko — APPROVED** [review]

**Review by Svetloslav15 — APPROVED** [review]

**Pull request merged**